chcemvediet/apps/inforequests/tests/test_views/test_detail.py

```diff
@@ -0,0 +1,286 @@
+        inforequest, _, _ = self._create_inforequest_scenario(u'clarification_request')
+        self._create_inforequest_email(inforequest=inforequest)
+        self._login_user()
+        response = self.client.get(reverse(u'inforequests:detail', args=(inforequest.slug, inforequest.pk)))
+        clarification_response_url = url(u'inforequests:clarification_response', branch=inforequest.main_branch)
+        self.assertTrue(inforequest.can_add_clarification_response)
+        self.assertNotContains(response, u'href="{}"'.format(clarification_response_url))
+
+    def test_inforequest_without_undecided_email_applicant_can_add_clarification_response(self):
+        inforequest, _, _ = self._create_inforequest_scenario(u'clarification_request')
+        self._login_user()
+        response = self.client.get(reverse(u'inforequests:detail', args=(inforequest.slug, inforequest.pk)))
+        clarification_response_url = url(u'inforequests:clarification_response', branch=inforequest.main_branch)
+        self.assertTrue(inforequest.can_add_clarification_response)
+        self.assertContains(response, u'href="{}"'.format(clarification_response_url))
+
+    def test_closed_inforequest_without_undecided_email_applicant_can_add_clarification_response(self):
+        inforequest, _, _ = self._create_inforequest_scenario(dict(closed=True), u'clarification_request')
+        self._login_user()
+        response = self.client.get(reverse(u'inforequests:detail', args=(inforequest.slug, inforequest.pk)))
+        clarification_response_url = url(u'inforequests:clarification_response', branch=inforequest.main_branch)
+        self.assertTrue(inforequest.can_add_clarification_response)
+        self.assertNotContains(response, u'href="{}"'.format(clarification_response_url))
+
+    def test_inforequest_with_undecided_email_applicant_can_add_appeal(self):
+        inforequest, _, _ = self._create_inforequest_scenario(u'advancement')
+        self._create_inforequest_email(inforequest=inforequest)
+        self._login_user()
+        response = self.client.get(reverse(u'inforequests:detail', args=(inforequest.slug, inforequest.pk)))
+        appeal_url = url(u'inforequests:appeal', branch=inforequest.main_branch)
+        self.assertTrue(inforequest.can_add_appeal)
+        self.assertNotContains(response, u'href="{}"'.format(appeal_url))
+
+    def test_inforequest_without_undecided_email_applicant_can_add_appeal(self):
+        inforequest, _, _ = self._create_inforequest_scenario(u'advancement')
+        self._login_user()
+        response = self.client.get(reverse(u'inforequests:detail', args=(inforequest.slug, inforequest.pk)))
+        appeal_url = url(u'inforequests:appeal', branch=inforequest.main_branch)
+        self.assertTrue(inforequest.can_add_appeal)
+        self.assertContains(response, u'href="{}"'.format(appeal_url))
+
+    def test_closed_inforequest_without_undecided_email_applicant_can_add_appeal(self):
```

Comment:
@viliambalaz Tu ti nesenia niektore nazvy testov. By som cakal, ze niektore testy budu `*_can_*` a niektore `*_cannot_*`. Teda aspon by som cakal, ze testujeme, ze nieco sa za nejakych okolnosti da a za nejakych inych okolnosti neda spravit. Teraz to vyzera, ako keby sa tie veci dali spravit vzdy.

Reply:
:ok: